frontend/src/pages/CollaborationPage.js

```diff
@@ -32,28 +48,30 @@ const CollaborationPage = () => {
 
   useEffect(() => {
     if (!location.state?.room) {
-      navigate(homeUrl, { replace: true })
+      setIsPartnerLeft(true)
       return
     }
-
     collabSocket.emit('join-room', getCollabRoomId(location.state.room))
     matchingSocket.emit('join-room', location.state.room)
   }, [location.state, navigate])
 
   useEffect(() => {
     collabSocket.on('partner-disconnected', () => {
+      // TODO: CHECK frontend again after matching service bug is rectified
       setIsPartnerOnline(false)
     })
     collabSocket.on('partner-connected', (roomClients) => {
-      // check if user is the only one in the room or if there are more ppl
+      // Check if user is the only one in the room or if there are more ppl
       setIsPartnerOnline(roomClients.length > 1)
     })
   }, [])
 
   useEffect(() => {
     matchingSocket.on('partner-left', (data) => {
       if (data === 'partner left') {
-        navigate(homeUrl)
+        // TODO: CHECK frontend again after matching service bug is rectified
+        setIsPartnerOnline(null)
```

Comment:
Why do we set this to null?

Reply:
When the partner left the room, it seems that it is a form of "disconnect". Hence, I set it as null to ensure that the `PartnerOfflineAlertDialog `does not appear together with the `PartnerLeftAlertDialog`. It seems to be working, but more rigorous testing is needed.

Reply:
Ohhhhhh okay I see I see... just a suggestion, but could we set it to `true` instead? Then the dialog still won't appear but it'll less confusing to others reading the code cos the intent is more obvious

Reply:
Yup I did think of setting it to `true`. Okay will change it!

Reply:
Anyway for this, I got a workaround for this. It is now working. I am adding comments in the code so you guys will understand.